### PR TITLE
Add route 53 to cloudfront

### PR DESCRIPTION
**Pull request by MancunianSam**
- Add create_hosted_zone variable.
- Add domain name alias and ACM certificate

### CODE REVIEW

File: cloudfront/outputs.tf [hunk 1]
```diff
@@ -1,3 +1,11 @@
 output "cloudfront_oai_iam_arn" {
   value = aws_cloudfront_origin_access_identity.origin_access_identity.iam_arn
 }
+
+output "cloudfront_domain_name" {
```
Comment (TomJKing) [inline cloudfront/outputs.tf:5]: Is it worth putting `s3` into the name of the output variable, just to make a little clearer?

File: cloudfront/outputs.tf [hunk 2]
```diff
@@ -1,3 +1,11 @@
...
+  value = aws_cloudfront_distribution.cloudfront_s3_distribution.domain_name
+}
+
+output "cloudfront_hosted_zone_id" {
```
Comment (TomJKing) [inline cloudfront/outputs.tf:9]: Is it worth putting `s3` into the name of the output variable, just to make a little clearer?

**Review by TomJKing — COMMENTED** [review]
Looks good. Couple of minor comments

File: cloudfront/outputs.tf (hunk 1 above)
Comment (MancunianSam) [reply cloudfront/outputs.tf:5]: It's not really the s3 domain name, it's the domain name of the cloudfront distribution. I think it's the name of the distribution resource that's confusing things. I named it before I realised it was doing more than just s3 requests. So I would say leave this as is and rename `cloudfront_s3_distribution` to `cloudfront_distribution` What do you think?

File: cloudfront/outputs.tf (hunk 2 above)
Comment (MancunianSam) [reply cloudfront/outputs.tf:9]: Same as for the comment above, this is the hosted zone id of the whole distribution.

File: cloudfront/outputs.tf (hunk 1 above)
Comment (TomJKing) [reply cloudfront/outputs.tf:5]: Sounds good to me

**Review by TomJKing — APPROVED** [review]

**Pull request merged**